src/nupic/algorithms/Connections.cpp

```diff
@@ -54,11 +54,10 @@ void Connections::initialize(CellIdx numCells,
                              SynapseIdx maxSynapsesPerSegment)
 {
   cells_ = vector<CellData>(numCells);
-  numSegments_ = 0;
-  numSynapses_ = 0;
-  nextFlatIdx_ = 0;
   maxSegmentsPerCell_ = maxSegmentsPerCell;
   maxSynapsesPerSegment_ = maxSynapsesPerSegment;
+  nextSegmentOrdinal_ = 0;
```

Comment:
Put a short comment before this line explaining that we just increment this each time we need an ordinal to keep track of the ordering of segments